### PR TITLE
[Core] Remove strategy ctor deprecation warnings

### PR DESCRIPTION
**Pull request by sunethwarna**
**Description**
Removes deprecations warnings coming from strategy ctors

**Changelog**
- Remove deprecation warnings from `ResidualBasedLinearStrategy`
- Remove deprecation warnings from `ResidualBasedNewtonRaphsonStrategy`
- Remove deprecation warnings from `LineSearchStrategy`

### CODE REVIEW

File: kratos/solving_strategies/strategies/line_search_strategy.h [hunk 1]
```diff
@@ -230,6 +280,7 @@ class LineSearchStrategy
      * @param pNewBuilderAndSolver The builder and solver employed
      * @param ThisParameters Settings used in the strategy
      */
+    KRATOS_DEPRECATED_MESSAGE("Constructor deprecated, please use the constructor without linear solver")
```
Comment (philbucher) [inline kratos/solving_strategies/strategies/line_search_strategy.h]: the CTors with Parameters are quite new, I think we can directly take the lin solver argument out without backwards compatibility

File: kratos/solving_strategies/strategies/line_search_strategy.h [hunk 2]
```diff
@@ -170,6 +170,34 @@ class LineSearchStrategy
...
+        typename TConvergenceCriteriaType::Pointer pNewConvergenceCriteria,
+        typename TBuilderAndSolverType::Pointer pNewBuilderAndSolver,
+        int MaxIterations = 30,
+        bool CalculateReactions = false,
+        bool ReformDofSetAtEachStep = false,
+        bool MoveMeshFlag = false
+        ) : BaseType(rModelPart, pScheme, pNewConvergenceCriteria, pNewBuilderAndSolver, MaxIterations, CalculateReactions, ReformDofSetAtEachStep, MoveMeshFlag)
+    {
+        const Parameters default_settings_const = this->GetDefaultParameters();
+        Parameters default_settings(default_settings_const);
```
Comment (philbucher) [inline kratos/solving_strategies/strategies/line_search_strategy.h]:
```suggestion
        Parameters default_settings = this->GetDefaultParameters();
```

File: kratos/solving_strategies/strategies/line_search_strategy.h (hunk 1 above)
Comment (sunethwarna) [reply kratos/solving_strategies/strategies/line_search_strategy.h]: done :)

File: kratos/solving_strategies/strategies/line_search_strategy.h (hunk 2 above)
Comment (sunethwarna) [reply kratos/solving_strategies/strategies/line_search_strategy.h]: done :)

**Review by philbucher — APPROVED** [review]
:+1:

**Pull request merged**